### ISSUE TITLE
More information about workloads

### ISSUE THREAD

**Issue by jacopopalumbo01** [mit-emze/cimloop#3]
@giorgiodaneri and myself are working on a university project about mapping various transformer architectures on different CiM implementations.
It would be nice to have more details about the workload structure, regarding the generation of the histograms in particular.

We've tried to rebuild the histograms of AlexNet as a starting point with the intent of checking if our implementation is consistent with yours. However we obtained different results (in particular in the weights histograms, which should be the same as yours).

- Which models did you use as reference? We've tried different pre-trained models but none is consistent with your results.
- Did you apply any sort of approximation/truncation after the histogram normalization?

Thanks

**Comment by tanner-andrulis** [comment]
For most of the CNNs, I used 8b-quantized models from the Torchvision library. I downloaded them and directly used the quantized values from the DNNs. In the process, I squished the histogram from 256 to 15 bins to make smaller YAML files.


For Transformers I used the original models from Huggingface and I clipped to the 98% interior activations before quantizing to 15 bins. All tensors are symmetrically quantized.

Beyond this, I made no other changes.

I would be cautious about trying to exactly reproduce results, since there are many copies of the same DNNs out there (and existing DNNs are being posted with different weights). Do the distributions generally match up? If so, then I would consider your results to be accurate.
